neonvm/controllers/virtualmachine_controller.go

```diff
@@ -1011,3 +1120,56 @@ func DeepEqual(v1, v2 interface{}) bool {
 
 	return reflect.DeepEqual(x1, x2)
 }
+
+func (r *VirtualMachineReconciler) tryUpdateVM(ctx context.Context, virtualmachine *vmv1.VirtualMachine) error {
+	// update vm (try 10 times)
+	var try int
+	var err error
+	for try = 1; try <= 10; try++ {
+		if err = r.Update(ctx, virtualmachine); err != nil {
+			if apierrors.IsConflict(err) {
+				time.Sleep(time.Second)
+				continue
+			}
+			return err
+		} else {
+			// vm was updated
+			break
+		}
+	}
+
+	if try > 10 {
+		return err
+	}
+
+	return nil
+}
```

Comment:
1. sorry, didn't get you. `r.Update` pass on first attempt usually. This trick (with tries) because reconcile function has `Requeue:true` somewhere. After redesign this part will be removed.
2. I'm not sure is a big difference between PUT and PATCH rest calls in term of performance. But with  PUT (r.Update) I ensure object will be exactly as I specified.

Reply:
added to follow ups list

Reply:
Respectfully, this can't be left to a follow-up — to do so would be unnecessarily sloppy, for no gain. I've provided an isolated reproduction of the function, demonstrating that not only is the function incorrect, but its current implementation is actively harmful compared to simpler alternatives. Switching to patch instead of update can be left for later, sure. But there's no excuse for using this function instead of `r.Update`, when it is strictly worse _and_ API-equivalent to `r.Update`.

Reply:
tryUpdateVM now just r.Update() call